airbyte-integrations/connectors/source-salesforce/source_salesforce/source.py

```diff
@@ -127,7 +134,22 @@ def streams(self, config: Mapping[str, Any]) -> List[Stream]:
         stream_objects = sf.get_validated_streams(config=config, catalog=self.catalog)
         streams = self.generate_streams(config, stream_objects, sf)
         streams.append(Describe(sf_api=sf, catalog=self.catalog))
-        return streams
+        # TODO: incorporate state & ConcurrentCursor when we support incremental
+        configured_streams = []
+        for stream in streams:
+            sync_mode = self._get_sync_mode_from_catalog(stream)
+            if sync_mode == SyncMode.full_refresh:
+                configured_streams.append(StreamFacade.create_from_stream(stream, self, logger, self.MAX_WORKERS, None, NoopCursor()))
+            else:
+                configured_streams.append(stream)
+        return configured_streams
+
+    def _get_sync_mode_from_catalog(self, stream: Stream) -> Optional[SyncMode]:
+        if self.catalog:
+            for catalog_stream in self.catalog.streams:
```

Comment:
Outside of the scope of this PR: I see this looping logic more and more. Would there be a way to have a `ConfiguredAirbyteCatalog.get_stream_by_name(name: str)`? This would greatly improve the readability/maintenance of our code